packages/jest-snapshot/src/types.ts

```diff
@@ -47,6 +54,19 @@ export interface SnapshotMatchers<R extends void | Promise<void>, T> {
     propertyMatchers: Partial<U>,
     hint?: string,
   ): R;
+  /**
+   * This ensures that a value matches the specific snapshot.
+   * Instead of use current test name in global state, it will use the specific name to find the snapshot.
+   */
+  toMatchNamedSnapshot(snapshotName: string): R;
+  /**
+   * This ensures that a value matches the specific snapshot with property matchers.
+   * Instead of use current test name in global state, it will use the specific name to find the snapshot.
+   */
+  toMatchNamedSnapshot<U extends Record<keyof T, unknown>>(
+    snapshot: string,
+    propertyMatchers: Partial<U>,
```

Comment:
If `propertyMatchers` would be marked optional, I think there shouldn’t be a need for overload. Or?

```suggestion
    snapshotName: string,
    propertyMatchers?: Partial<U>,
```